source/common/http/http2/metadata_decoder.cc

```diff
@@ -27,14 +65,22 @@ bool MetadataDecoder::receiveMetadata(const uint8_t* data, size_t len) {
 }
 
 bool MetadataDecoder::onMetadataFrameComplete(bool end_metadata) {
-  bool success = decodeMetadataPayloadUsingNghttp2(end_metadata);
+  bool success;
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.http2_decode_metadata_with_quiche")) {
+    success = decodeMetadataPayload(end_metadata);
+  } else {
+    success = decodeMetadataPayloadUsingNghttp2(end_metadata);
+  }
   if (!success) {
     return false;
   }
 
   if (end_metadata) {
     callback_(std::move(metadata_map_));
     metadata_map_ = std::make_unique<MetadataMap>();
+    decoder_context_ =
```

Comment:
This is the same call as in the constructor, right? I wonder if it would be worth a 1-line initializeHpackDecoderContext() method?
```
void MetadataDecoder::initializeHpackDecoderContext() {
  decoder_context_ =  std::make_unique<HpackDecoderContext>(*metadata_map_, max_payload_size_bound_);
}
```
And then use it in both places? Actually, you could initialize metadata_map_ in the same method which would also be nice.

Reply:
Done, thanks for the suggestion.